frontend/scripts/metrica.js

```diff
@@ -0,0 +1,319 @@
+(() => {
+    class requestService {
+        post(url, data, headers = {}) {
+            return fetch(url, {
+                method: 'POST',
+                crossDomain: true,
+                headers: headers,
+                body: data
+            });
+        }
+    }
+    const state = {
+        host_api: 'http://stage.metrica.fun/api/v1',
+        host: 'http://stage.metrica.fun/',
+        routes: {
+            create_visitor: '/visitors',
+            create_visit: '/visits'
+        }
+    };
+    window._metricaTracking = {
+        endTime:undefined,
+        configMetrica:undefined,
+        initialize() {
+            this.setObjectMetricaConf();
+            let token = this.getToken();
+            if (token) {
+                this.createVisit();
+            } else {
+                this.createVisitor(this.getTrackById())
+                    .finally(() => this.createVisit());
+            }
+
+        },
+        setObjectMetricaConf() {
+            let metricaConfig = Helper.getPropByString(window, '_metricaTrackingConfig');
+            if(metricaConfig === undefined) {
+                window._metricaTrackingConfig = [];
+            }
+            if(metricaConfig[0][0] !== 'dateStart' || metricaConfig[0][1] === undefined) {
```

Comment:
if condition above is true, this string will break script:

metricaConfig = undefined;
metricaConfig[0] - Uncaught exception

make it safety